#14 get all tasks

DraganKec opened the pull request.

Commit:
#14 Create get all tasks

Commit:
#14 Change two method getTasks for today and tomorrow to one method getTasksByCreatedAt

cctraining commented on src/main/java/ba/codecentric/prioritytasks/controller/TaskController.java:
```diff
@@ -3,24 +3,44 @@
+    public List<Task> getByCreatedAt(@PathVariable @DateTimeFormat(pattern="yyyy-MM-dd") Date date) throws Exception {
+
+        List<Task> tasks = taskService.getByCreatedAt(date);
+        if (tasks.size() == 0) throw new Exception("No more tasks");
```
I'm more of a curly braces guy for the if statement :)

cctraining replied:
tasks is list, so you should use `tasks.isEmpty()`

cctraining commented on src/main/java/ba/codecentric/prioritytasks/controller/TaskController.java:
```diff
@@ -3,24 +3,44 @@
+    }
+
+    @GetMapping(value = "/tasks/{date}")
+    public List<Task> getByCreatedAt(@PathVariable @DateTimeFormat(pattern="yyyy-MM-dd") Date date) throws Exception {
```
Can this be also `getAllTasks(...)` instead of `getByCreatedAt(...)`

Commit:
#14 Change getByCreatedAt to getAllTasks
#14 Add curly braces to if statement in getAllTasks
#14 Change List.size to List.IsEmpty in GetAllTasks

cctraining commented on src/main/java/ba/codecentric/prioritytasks/service/TaskService.java:
```diff
@@ -15,7 +18,15 @@ public TaskService(TaskRepository taskRepository) {
         return taskRepository.save(task);
     }
+
+    public List<Task> getAllTask() {
```
`getAllTask` -> `getAllTasks`

cctraining commented on src/main/java/ba/codecentric/prioritytasks/service/TaskService.java:
```diff
@@ -15,7 +18,15 @@ public TaskService(TaskRepository taskRepository) {
+        return taskRepository.findAll();
+    }
+
+    public List<Task> getByCreatedAt(Date createdAt) {
```
`getByCreatedAt(...)` -> `getAllTasks(...)`

Commit:
#14 Change getByCreatedAt to getAllTasks in TaskRepository and TaskService

nikola-lisac commented:
Avoid @Query if your query is simple enough to be written as a query method, spring data will create the query for you.
https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.query-creation
The method should get all tasks for the date, not just uncompleted ones.

Commit:
#14 Remove @Query annotation from tast repository and rename method to findByCreatedAt
